service/src/main/java/uk/nhs/adaptors/gpc/consumer/filters/SearchForDocumentsGatewayFilterFactory.java

```diff
@@ -0,0 +1,76 @@
+package uk.nhs.adaptors.gpc.consumer.filters;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.OrderedGatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+import uk.nhs.adaptors.gpc.consumer.utils.FindAPatientDocsUtil;
+
+@Component
+@Slf4j
+public class SearchForDocumentsGatewayFilterFactory extends AbstractGatewayFilterFactory<SearchForDocumentsGatewayFilterFactory.Config> {
+    private static final int PRIORITY = -3;
+    @Value("${gpc-consumer.gpc.gpcConsumerUrl}")
+    private String gpcConsumerUrl;
+    @Value("${gpc-consumer.gpc.gpcUrl}")
+    private String gpcUrl;
+    @Value("${gpc-consumer.gpc.findSearchDocumentsPath}")
+    private String findSearchDocumentsPath;
+    @Value("${gpc-consumer.gpc.structuredPath}")
+    private String structuredPath;
+    @Value("${gpc-consumer.gpc.findPatientPath}")
+    private String findPatientPath;
+
+
+    public SearchForDocumentsGatewayFilterFactory() {
+        super(SearchForDocumentsGatewayFilterFactory.Config.class);
+    }
+
+    @Override
+    public GatewayFilter apply(Config config) {
+        return new OrderedGatewayFilter((exchange, chain) ->
+            chain.filter(exchange.mutate().response(exchange.getResponse()).build()), PRIORITY);
+    }
+
+    @Bean
+    public RouteLocator routes(RouteLocatorBuilder builder) {
```

Comment:
I'm not comfortable with defining our routes twice - once in a YAML and again in Java. Since documentation says we can only configure modifyResponseBody I think we'll need to define everything in Java config.

In that case this class name SearchForDocumentsGatewayFilterFactory.java  is misleading - since this class defines the all of the routes for the application.

Reply:
Yaml config removed